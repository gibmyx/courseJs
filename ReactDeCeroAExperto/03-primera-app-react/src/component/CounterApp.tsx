import {useState} from "react";

function CounterApp({value} : {value: number}) {

    const [count, setCount] = useState(value)

    const handleAdd = () => setCount(prevState => prevState +1 )
    const handleSubtract = () => setCount(prevState => prevState -1 )
    const handleReset = () => setCount(value)

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{count}</h2>
            <button onClick={handleAdd}>+1</button>
            <button onClick={handleReset}>Reset</button>
            <button onClick={handleSubtract}>-1</button>
        </>
    )
}

export default CounterApp