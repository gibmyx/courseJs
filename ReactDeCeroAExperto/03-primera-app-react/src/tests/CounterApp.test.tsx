import React from "react";
import {beforeEach, describe, expect, test} from "vitest";
import {shallow} from "enzyme";
import CounterApp from "../component/CounterApp";

describe('Pruebas en <CounterApp/>', () => {
    let wrapper = shallow(<CounterApp value={100}/>);
    beforeEach(() => {
        wrapper = shallow(<CounterApp value={100}/>);
    })

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('El valor por defaulf debe ser 100', () => {
        const wrapper = shallow(<CounterApp value={100}/>)
        const valorcounter = wrapper.find('h2').text().trim();
        expect(valorcounter).toBe('100');
    })

    test('Debe de incrementar con el Boton +1', () => {
        console.log( wrapper.find('button'))
        wrapper.find('button').at(0).simulate('click');
        const valorcounter = wrapper.find('h2').text().trim();
        expect(valorcounter).toBe('101');
    })

    test('Debe de decrementar con el Boton -1', () => {
        wrapper.find('button').at(2).simulate('click');
        const valorcounter = wrapper.find('h2').text().trim();
        expect(valorcounter).toBe('99');
    })

    test('Debe de colocar el valor por defecto con el btn rest', () => {
        const wrapper = shallow(<CounterApp value={200}/>)

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        const valorCounter = wrapper.find('h2').text().trim();
        expect(valorCounter).toBe('202');

        wrapper.find('button').at(1).simulate('click');
        const valorCounterReset = wrapper.find('h2').text().trim();
        expect(valorCounterReset).toBe('200');
    })
})