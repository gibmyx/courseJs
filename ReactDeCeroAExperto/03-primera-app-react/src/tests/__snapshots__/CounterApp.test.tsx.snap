// Vitest Snapshot v1

exports[`Pruebas en <CounterApp/> > Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h1>
    CounterApp
  </h1>
  <h2>
    100
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
